learnyounode/09-time-server.js

```diff
@@ -0,0 +1,16 @@
+var net = require('net');
+
+function pad(n) { return n < 10 ? '0' + n : n }
+
+var server = net.createServer(function(socket) {
+  d = new Date();
+  s = d.getFullYear() + "-"
+    + pad(d.getMonth() + 1) + "-"
+    + pad(d.getDate()) + " "
+    + pad(d.getHours()) + ":"
+    + pad(d.getMinutes()) + "\n";
+  socket.end(s);
+});
+server.listen(process.argv[2]);
+
+//Struggled with this one. Solution from https://github.com/frasertweedale/learnyounode-solutions/blob/master/time.js
```

Comment:
The idea with this one is that we aren't even sending HTTP messages anymore, just plain old text (on top of TCP). `pad` is used to add extra zeros to a number so that all numeric strings have two digits. The callback given to `.createServer` accepts a socket as an argument, and calls `socket.end` to actually send out a response.